src/web-ui/src/router/index.js

```diff
@@ -57,124 +52,135 @@ function getCognitoUser() {
+        const { data } = await UsersRepository.getUserByUsername(cognitoUser.username)
+        storeUser = data
+      }
+
+      const credentials = await Credentials.get();
+
+      if (!storeUser.id) {
+        // Store user does not exist. Create one on the fly.
+        // This takes the personalize User ID which was a UUID4 for the current session and turns it into a user user ID.
+        console.log('store user does not exist for cognito user... creating on the fly')
+        let identityId = credentials ? credentials.identityId : null;
+        let provisionalUserId = AmplifyStore.getters.personalizeUserID;
+        const { data } = await UsersRepository.createUser(provisionalUserId, cognitoUser.username, cognitoUser.attributes.email, identityId)
+        storeUser = data
+      }
+
+      console.log('Syncing store user state to cognito user custom attributes')
+      // Store user exists. Use this as opportunity to sync store user
+      // attributes to Cognito custom attributes.
+      Auth.updateUserAttributes(cognitoUser, {
+        'custom:profile_user_id': storeUser.id.toString(),
+        'custom:profile_email': storeUser.email,
+        'custom:profile_first_name': storeUser.first_name,
+        'custom:profile_last_name': storeUser.last_name,
+        'custom:profile_gender': storeUser.gender,
+        'custom:profile_age': storeUser.age.toString(),
+        'custom:profile_persona': storeUser.persona
+      })
 
-    console.log('Syncing store user state to cognito user custom attributes')
-    // Store user exists. Use this as opportunity to sync store user
-    // attributes to Cognito custom attributes.
-    await Vue.prototype.$Amplify.Auth.updateUserAttributes(cognitoUser, {
-      'custom:profile_user_id': storeUser.id.toString(),
-      'custom:profile_email': storeUser.email,
-      'custom:profile_first_name': storeUser.first_name,
-      'custom:profile_last_name': storeUser.last_name,
-      'custom:profile_gender': storeUser.gender,
-      'custom:profile_age': storeUser.age.toString(),
-      'custom:profile_persona': storeUser.persona
-    })
-
-    // Sync identityId with user to support reverse lookup.
-    if (credentials && storeUser.identity_id != credentials.identityId) {
-      console.log('Syncing credentials identity_id with store user profile')
-      storeUser.identity_id = credentials.identityId
-    }
+      // Sync identityId with user to support reverse lookup.
+      if (credentials && storeUser.identity_id != credentials.identityId) {
+        console.log('Syncing credentials identity_id with store user profile')
+        storeUser.identity_id = credentials.identityId
+      }
 
-    // Update last sign in and sign up dates on user.
-    let newSignUp = false
+      // Update last sign in and sign up dates on user.
+      let newSignUp = false
 
-    const now = new Date()
-    storeUser.last_sign_in_date = now.toISOString()
+      const now = new Date()
```

Comment:
Looks like some extra whitespace has been added to the start of the lines here, can you remove it?